src/WaterLily.jl

```diff
@@ -69,9 +69,11 @@ mutable struct Simulation <: AbstractSimulation
                         T=Float32, mem=Array) where N
         @assert !(isa(u_BC,Function) && isa(uλ,Function)) "`u_BC` and `uλ` cannot be both specified as Function"
         @assert !(isnothing(U) && isa(u_BC,Function)) "`U` must be specified if `u_BC` is a Function"
-        isa(u_BC,Function) && @assert all(typeof.(ntuple(i->u_BC(i,zero(T)),N)).==T) "`u_BC` is not type stable"
-        uλ = isnothing(uλ) ? ifelse(isa(u_BC,Function),(i,x)->u_BC(i,0.),(i,x)->u_BC[i]) : uλ
-        U = isnothing(U) ? √sum(abs2,u_BC) : U # default if not specified
+        isa(g,Function) && @assert first(methods(g)).nargs==4 "g::Function needs to be defined as g(i,x,t)"
```

Comment:
Should we also check the type stability of `g`? Same as `u_BC`.

Reply:
For the accelerating flow test, `g(i,x,t) = i==1 ? t*jerk : zero(T)` is not possible in GPU, as we have seen before. So I have added just the Float64 test, `g(i,x,t) = i==1 ? t*jerk : 0.`. Users will need to be weary of the precision they are using when defining these functions.